docs/source/tutorial/type_check.rst

```diff
@@ -281,9 +280,9 @@ And then, get each type:
 
    x_type, y_type = in_types
 
-Don't get each value before check ``in_types.size()``.
-When the number of argument is illegal, this process may fail.
-For example, this code doesn't work when the size of ``in_types`` is zero:
+Don't get each value before checking ``in_types.size()``.
+When the number of argument is illegal, this will fail.
```

Comment:
I think the wording "may fail" is appropriate because there are some situations `type_check.expect` does not "fail", or, `type_check.expect` can correctly throw an appropriate error even if the number of argument of `__call__` is incorrect.

For example, suppose the expected number of argument is 2 and `ndim` of first argument is 3 (this is same example as the tutorial). If the number of argument is 0, `type_check.expect` will "fail", or it emits strange error message. But if the number of argument is 3, it does not "fail", or, `type_check.expect` can correctly throw `InvalidType` error. 

I think the meaning of `fail` here is difficult to understand. How about changing the description to "When the number of argument is illegal, `type_check.expect` can output strange error messages"?